### ISSUE TITLE
Align currently supported Node.js version

### ISSUE THREAD

**Issue by nikeee** [parse-community/Parse-SDK-JS#2065]
### New Issue Checklist
- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases).

### Issue Description
According to the readme, parse follows the supported node versions:

> Parse JS SDK is continuously tested with the most recent releases of Node.js to ensure compatibility.
> We follow the [Node.js Long Term Support plan](https://github.com/nodejs/Release) and only
> test against versions that are officially supported and have not reached their end-of-life date.

We should align the repository with the currently supported Node.js versions: https://github.com/nodejs/Release

### Steps to reproduce
Install parse when using Node.js 20 or 21. Get an engine warning:
```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'parse@4.3.1',
npm WARN EBADENGINE   required: { node: '>=14.21.0 <17 || >=18 <20' },
npm WARN EBADENGINE   current: { node: 'v21.4.0', npm: '10.2.1' }
npm WARN EBADENGINE }
```

### Expected Outcome
According to the readme, no warning should occur.

###  Environment
- Node: `v21.4.0`
- `parse@4.3.1`

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-alpha.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.0.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.0.0)
